batch/batches/Storage/Periodic/KAsyncStoragePeriodicExport.class.php

```diff
@@ -0,0 +1,267 @@
+		$storageExportData->srcFileEncryptionKey = $fileSync->srcEncKey;
+		$storageExportData->srcFileSyncId = $fileSync->id;
+
+		$storageExportData->force = $force;
+		$storageExportData->destFileSyncStoredPath = $externalStorage->storageBaseDir . '/' . $fileSync->filePath;
+		$storageExportData->createLink = $externalStorage->createFileLink;
+
+		if($externalStorage->protocol == StorageProfileProtocol::S3)
+		{
+			$storageExportData = $this->addS3FieldsToStorageData($storageExportData, $externalStorage);
+		}
+
+		return $storageExportData;
+	}
+
+	protected function addS3FieldsToStorageData($storageExportData, $externalStorage)
+	{
+		$storageExportData->filesPermissionInS3 = $externalStorage->filesPermissionInS3;
+		$storageExportData->s3Region = $externalStorage->s3Region;
+		$storageExportData->sseType = $externalStorage->sseType;
+		$storageExportData->sseKmsKeyId = $externalStorage->sseKmsKeyId;
+		$storageExportData->signatureType = $externalStorage->signatureType;
+		return $storageExportData;
+	}
+
+	protected function initResponseProfile()
+	{
+		$responseProfile = new KalturaDetachedResponseProfile();
+		$responseProfile->type = KalturaResponseProfileType::INCLUDE_FIELDS;
+		$responseProfile->fields = self::RESPONSE_PROFILE_FIELDS;
+		return $responseProfile;
+	}
+
+	protected function getStorageProfile()
+	{
+		$storageProfileIdsArray = explode(',', $this->getAdditionalParams(self::STORAGE_PROFILE_IDS));
+		if($this->currentIndex >= count($storageProfileIdsArray))
+		{
+			$this->currentIndex = 0;
+		}
+
+		if(isset($this->storageProfiles[$this->currentIndex]))
+		{
+			$index = $this->currentIndex;
+			$id = $this->storageProfiles[$index]->id;
+			KalturaLog::debug("Using Storing profile [$id], index [$this->currentIndex]");
+			$this->currentIndex++;
+			return $this->storageProfiles[$index];
+		}
+
+		$storageProfileId = $storageProfileIdsArray[$this->currentIndex];
+
+		$responseProfile = new KalturaDetachedResponseProfile();
+		self::$kClient->setResponseProfile($responseProfile);
+		$storageProfile = self::$kClient->storageProfile->get($storageProfileId);
+		if (!$storageProfile)
+		{
+			KalturaLog::debug("Could not find storage Profile id [$storageProfileId]");
+		}
+
+		KalturaLog::debug("Storing profile [$storageProfileId] info, index [$this->currentIndex]");
+		$this->storageProfiles[$this->currentIndex] = $storageProfile;
+		$this->currentIndex++;
+		return $storageProfile;
+	}
+
+	protected function exportFileSyncs($fileSyncs, $storageProfile)
+	{
+		foreach ($fileSyncs as $fileSync)
+		{
+			KalturaLog::debug("Export file sync id [{$fileSync->id}] to storage id [{$storageProfile->id}]");
+			try
+			{
+				$exported = $this->export($storageProfile, $fileSync);
+				if($exported)
+				{
+					$status = KalturaFileSyncStatus::READY;
+				}
+				else
+				{
+					$status = KalturaFileSyncStatus::ERROR;
+				}
+				$this->changeFileSyncStatus($fileSync, $status);
```

Comment:
Since there file sync status will happen many times - use multi request for optimisation